predicators/nsrt_learning/strips_learning/base_strips_learner.py

```diff
@@ -260,6 +260,19 @@ def _find_best_matching_pnad_and_sub(
                        not segment.necessary_add_effects.issubset(
                            ground_op.add_effects):
                         continue
+                    keep_effects = ground_op.preconditions & \
```

Comment:
it shouldn't affect the old results, since in the `BackchainingSTRIPSLearner`, we don't change the `segment.necessary_image`, but you're right that this is probably a good thing to do.